### PR TITLE
docs: explain defaultRoute

### PR DESCRIPTION
**Pull request by aradwann**
#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

closes: Issue #3911

### CODE REVIEW

**Review by Fdawgs — CHANGES_REQUESTED** [review]
Some spelling and grammar issues, will properly review when i get the chance.

**Review by mcollina — APPROVED** [review]
lgtm

File: docs/Reference/Server.md [hunk 1]
```diff
@@ -961,6 +963,8 @@ const defaultRoute = fastify.getDefaultRoute()
 #### setDefaultRoute
 <a id="setDefaultRoute"></a>
 
+**Important Note**: If you override the default route, it means that the 404 router will never be triggered. That also means `setNotFoundHandler` will not work, and no hooks fired. thus, if you want to customize handling 404s you can use `setNotFoundHandler` instead
+
```
Comment (Fdawgs) [inline docs/Reference/Server.md:967]:
```suggestion
**Note**: The default 404 handler, or one set using `setNotFoundHandler`, will never trigger if the default route is overridden. Use [setNotFoundHandler](#setnotfoundhandler) if you want to customize 404 handling instead.
```

File: docs/Reference/Server.md [hunk 2]
```diff
@@ -952,6 +952,8 @@ Note that the array contains the `fastify.server.address()` too.
 #### getDefaultRoute
 <a id="getDefaultRoute"></a>
 
+`defaultRoute` is a handler that handles the requests that do not match any URL specified inside your application which defaults to 404 router but can be overridden with [setDefaultRoute](#setdefaultroute)
+
```
Comment (Fdawgs) [inline docs/Reference/Server.md:956]:
```suggestion
The `defaultRoute` handler handles requests that do not match any URL specified by your Fastify application. This defaults to the 404 handler, but can be overridden with [setDefaultRoute](#setdefaultroute).
```

**Review by Fdawgs — CHANGES_REQUESTED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

File: docs/Reference/Server.md [hunk 3]
```diff
@@ -952,6 +952,7 @@ Note that the array contains the `fastify.server.address()` too.
 #### getDefaultRoute
 <a id="getDefaultRoute"></a>
 
+The `defaultRoute` handler handles requests that do not match any URL specified by your Fastify application. This defaults to the 404 handler, but can be overridden with [setDefaultRoute](#setdefaultroute).
```
Comment (jsumners) [inline docs/Reference/Server.md:955]: Please hard wrap lines to be consistent with the rest of the document.

File: docs/Reference/Server.md [hunk 4]
```diff
@@ -961,6 +962,7 @@ const defaultRoute = fastify.getDefaultRoute()
 #### setDefaultRoute
 <a id="setDefaultRoute"></a>
 
+**Note**: The default 404 handler, or one set using `setNotFoundHandler`, will never trigger if the default route is overridden. Use [setNotFoundHandler](#setnotfoundhandler) if you want to customize 404 handling instead.
```
Comment (jsumners) [inline docs/Reference/Server.md:965]: Please hard wrap lines to be consistent with the rest of the document.

File: docs/Reference/Server.md (hunk 3 above)
Comment (aradwann) [reply docs/Reference/Server.md]: done

File: docs/Reference/Server.md (hunk 4 above)
Comment (aradwann) [reply docs/Reference/Server.md]: done

**Review by RafaelGSS — APPROVED** [review]

**Pull request merged**